troposphere/static/js/actions/APITokenActions.js

```diff
@@ -0,0 +1,71 @@
+import APITokenConstants from "constants/APITokenConstants";
+import APIToken from "models/APIToken";
+import NotificationController from "controllers/NotificationController";
+import Utils from "./Utils";
+
+export default {
+    create: (name, userId) => {
+        if (!name) throw new Error("Missing Token name");
+        if (!userId) throw new Error("Missing Token author");
+        let apiToken = new APIToken({
+            name,
+            atmo_user: userId
+        });
+
+        // Add token optimistically
+        Utils.dispatch(APITokenConstants.ADD_TOKEN, {apiToken});
+
+        let promise = Promise.resolve(apiToken.save());
+        promise.then(() => {
+            Utils.dispatch(APITokenConstants.UPDATE_TOKEN, {apiToken});
+        });
+        promise.catch(() => {
+            Utils.dispatch(APITokenConstants.REMOVE_TOKEN, {apiToken});
+            NotificationController.error(
+                "Error creating token.",
+                "Your login might be expired. If you continue to see this error " +
```

Comment:
Notification controller needs to include the error from the create response. Take a look at the arg to catch.

Reply:
I can add this change